### PR TITLE
add addon check in user-server

### PR DESCRIPTION
**Pull request by xuezhaojun**
Signed-off-by: xuezhaojun <zxue@redhat.com>

### CODE REVIEW

File: pkg/hub/user_proxy.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+	// parse clusterID from current requestURL
+	clusterID, kubeAPIPath, err := parseRequestURL(req.RequestURI)
+	if err != nil {
+		http.Error(wr, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// check if cluster-proxy addon installed in target managed cluster
+	_, err = u.addonClient.AddonV1alpha1().ManagedClusterAddOns(clusterID).Get(context.TODO(), config.ADDON, v1.GetOptions{})
```
Comment (skeeey) [inline pkg/hub/user_proxy.go:85]: you may also check the status of addon

File: pkg/hub/user_proxy.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+	return nil
+}
+
+// parseRequestURL
+// Example Input: <service-ip>:8080/<clusterID>/api/pods?timeout=32s
+// Example Output:
+// 	clusterID: <clusterID>
+// 	kubeAPIPath: api/pods
+func parseRequestURL(requestURL string) (clusterID string, kubeAPIPath string, err error) {
```
Comment (skeeey) [inline pkg/hub/user_proxy.go:192]: We may put this in the helpers

Comment (xuezhaojun) [reply pkg/hub/user_proxy.go]: Sure, added condition check.

File: pkg/hub/user_proxy.go [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+	// check if cluster-proxy addon installed in target managed cluster
+	clusterProxyAddon, err := u.addonClient.AddonV1alpha1().ManagedClusterAddOns(clusterID).Get(context.TODO(), config.ADDON, v1.GetOptions{})
+	if err != nil {
+		http.Error(wr, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	var manifestApplied = false
+	for _, c := range clusterProxyAddon.Status.Conditions {
```
Comment (skeeey) [inline pkg/hub/user_proxy.go:93]: you can use `meta.IsStatusConditionTrue` instead of this

Comment (xuezhaojun) [reply pkg/hub/user_proxy.go:93]: OK, got it

Comment (xuezhaojun) [reply pkg/hub/user_proxy.go]: Using IsStatusConditionFalse to check error

File: pkg/helpers/helpers.go [hunk 4]
```diff
@@ -11,3 +15,20 @@ func GetCurrentNamespace(defaultNamespace string) string {
 	}
 	return string(nsBytes)
 }
+
+// ParseRequestURL
+// Example Input: <service-ip>:8080/<clusterID>/api/pods?timeout=32s
+// Example Output:
+// 	clusterID: <clusterID>
+// 	kubeAPIPath: api/pods
+func ParseRequestURL(requestURL string) (clusterID string, kubeAPIPath string, err error) {
```
Comment (skeeey) [inline pkg/helpers/helpers.go:24]: In the future, we need add UT for this

Comment (xuezhaojun) [reply pkg/helpers/helpers.go:24]: Sure

**Pull request merged**